### ISSUE TITLE
Unable to fully extract `libobjc.A.dylib` from iOS 17 dyld_shared_cache

### ISSUE THREAD

**Issue by frankschlegel** [arandomdev/DyldExtractor#58]
**Describe the bug**
When trying to extract the frameworks from the iOS 17 beta shared cache, I noticed that `dyldex` fails to fully extract `libobjc.A.dylib`, which causes issues down the line when trying to generate runtime headers later.

It seems `dyldex` can't find the dependency `libobjc-env.dylib` inside the shared cache:

```
Processed: libobjc.A.dylib
----- libobjc.A.dylib -----
14:41:15:803 [ WARNING ] stub_fixer.py:181 : Unable to find dependency: b'/usr/local/lib/libobjc-env.dylib\x00'
--------------------
```

**To Reproduce**
1. iOS version: **17 beta 3**
2. DYLD target device and identifier, i.e iPhone X 11,2: **iPhone15,3**
3. Terminal command to reproduce: `dyldex -e libobjc.A.dylib dyld_shared_cache_arm64e`

**Comment by arandomdev** [comment]
Some images are linked to a frameworks outside the cache, but these usually don't affect the output. What kinds of issues are you facing when generating the runtime headers? What program are you using?

**Comment by arandomdev** [comment]
I believe this is fixed with 808a5ad. I was incorrectly getting the image index which is used to lookup the specific method list to use.
CIImage and CIFilter now have methods, and I was able to open them with ktool in GUI mode.

Please let me know if there is anything else.

**Comment by frankschlegel** [comment]
Yes, it works perfectly now. Thank you so much!
